### ISSUE TITLE
Iterating over NeighborLoader returns more nodes than in dataset

### ISSUE THREAD

**Issue by choosehappy** [pyg-team/pytorch_geometric#7021]
### 🐛 Describe the bug

I'm trying to walk my entire set of nodes using NeighborLoader to extract the DL feature vector....after spending a few hours on this i just can't figure out what I'm missing

This minimal example shows the issue:
1. i check how many nodes are in my dataset, 238k
2. I would thus expect NeighborLoader to return 238k of them, in an unshuffled manner (shuffle=False), so that i end up with a single feature vector per node. 
3. When i do this however, i end up with 245k feature vectors (note that no model is being applied below)

```python
print(train_data.num_nodes)
--238817

neighborloader = NeighborLoader(train_data, batch_size=2048, shuffle=False, 
                                num_neighbors=[-1, -1], num_workers=8, directed=True,  persistent_workers=True) 

xs= []
for data in tqdm.tqdm(neighborloader):
    xs.append(data.x.detach().cpu())
xs=torch.cat(xs, dim=0)

print(xs.shape)
--torch.Size([245603, 64])
```

Digging much deeper, i see that the same ~6k nodes are being computed upon twice. ,Sometimes the resulting feature vector is the same, other times wildly different, so even if i wanted to select one of the duplicates, its not clear to me which one is "more correct"

My graph is essentially a nearest neighbor based on radius in the euclidian space, so it yields an undirected graph (i.e., if 2 nodes are connected, they are both within a radius K and thus have an edge in both directions) 

Any thoughts would be greatly appreciated!

### Environment

* PyG version: master branch version (see below)
* PyTorch version: 1.13
* OS: WSL linux 
* Python version:  3.11
* CUDA/cuDNN version: 11.7
* How you installed PyTorch and PyG (`conda`, `pip`, source): pip
pyg-lib==0.2.0+pt113cu117
torch==1.13.0
torch-cluster==1.6.0+pt113cu117
torch-geometric @ git+https://github.com/pyg-team/pytorch_geometric.git@f0c72186286f257778c1d9293cfd0d35472d30bb
torch-optimizer==0.3.0
torch-scatter==2.1.0+pt113cu117
torch-sparse==0.6.16+pt113cu117

**Comment by choosehappy** [comment]
To answer my own question, they do appear to be in order:

```python
gids=[]
for data in tqdm.tqdm(neighborloader):
    gids.append(data.n_id.cpu()[0:len(data.input_id)])
    gids = np.asarray(torch.cat(gids, dim=0))


print(np.mean(gids==range(train_data.x.shape[0])))
---1.0
```

but i don't know if this is deterministically done, such that it can be relied upon in future versions, or if it this is happenstance :)

**Comment by rusty1s** [comment]
Seed nodes are always placed first in the batch, so you can get the node features of original node features via
```python
batch.x[:batch.batch_size]

**Comment by choosehappy** [comment]
Perfect, thanks!
